server/php-queries-types.cpp

```diff
@@ -0,0 +1,197 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2021 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#include <poll.h>
+
+#include "common/kprintf.h"
+#include "server/php-queries-types.h"
+#include "server/php-worker.h"
+#include "server/php-runner.h"
+#include "server/php-engine.h"
+#include "server/php-mc-connections.h"
+#include "server/php-sql-connections.h"
+
+void php_query_x2_t::run(php_worker *worker __attribute__((unused))) noexcept {
+  query_stats.desc = "PHPQX2";
+
+  php_script_query_readed(php_script);
+
+  //  worker->conn->status = conn_wait_net;
+  //  worker->conn->pending_queries = 1;
+
+  static php_query_x2_answer_t res;
+  res.x2 = val * val;
+
+  ans = &res;
+
+  php_script_query_answered(php_script);
+}
+
+void php_query_rpc_answer::run(php_worker *worker) noexcept {
+  query_stats.desc = "RPC_ANSWER";
+
+  if (worker->mode == rpc_worker) {
+    connection *c = worker->conn;
+    int *q = (int *)data;
+    int qsize = data_len;
+
+    vkprintf (2, "going to send %d bytes as an answer [req_id = %016llx]\n", qsize, worker->req_id);
+    send_rpc_query(c, q[2] == 0 ? TL_RPC_REQ_RESULT : TL_RPC_REQ_ERROR, worker->req_id, q, qsize);
+  }
+  php_script_query_readed(php_script);
+  php_script_query_answered(php_script);
+}
+
+void php_query_connect_t::run(php_worker *worker __attribute__((unused))) noexcept {
+  query_stats.desc = "CONNECT";
+
+  php_script_query_readed(php_script);
+
+  static php_query_connect_answer_t res;
+
+  switch (protocol) {
+    case protocol_type::memcached:
+      res.connection_id = get_target(host, port, &memcache_ct);
+      break;
+    case protocol_type::mysqli:
+      res.connection_id = sql_target_id;
+      break;
+    case protocol_type::rpc:
+      res.connection_id = get_target(host, port, &rpc_ct);
+      break;
+    default:
+      assert ("unknown protocol" && 0);
+  }
+
+  ans = &res;
+
+  php_script_query_answered(php_script);
+}
+
+void php_query_wait_t::run(php_worker *worker) noexcept {
+  query_stats.desc = "WAIT_NET";
+  php_script_query_readed(php_script);
+  php_script_query_answered(php_script);
+
+  php_worker_wait(worker, timeout_ms);
+}
+
+
+int php_worker_http_load_post_impl(php_worker *worker, char *buf, int min_len, int max_len) {
+  assert (worker != nullptr);
```

Comment:
This assert is not excessive? Or at least, should be moved to `php_worker_run_query`

Reply:
would be even better to pass php_worker type by reference.

Reply:
> 
moved assert to `php_worker_main`

Reply:
> 
php_worker * actually should be `this`, and all these functions should be methods of `php_worker` class. But let's leave it for separate PR